pkg/registry/core/service/storage/rest.go

```diff
@@ -876,3 +890,80 @@ func collectServiceNodePorts(service *api.Service) []int {
 	}
 	return servicePorts
 }
+
+func (rs *REST) getNetworkServiceIPAlloc(service *api.Service) (ipallocator.Interface, error) {
+	netName, ok := service.Labels[arktosv1.NetworkLabel]
+	if !ok {
+		netName = arktosv1.NetworkDefault
+	}
+
+	network, err := rs.getNetwork(genericapirequest.NewContext(), service.Tenant, netName)
+	if err != nil {
+		// todo: remove the temporary measure when we have all the multi-tenant networking components in place
+		// as temporary measure, ignore the error of default network not found
+		if netName == arktosv1.NetworkDefault {
+			return rs.serviceIPs, nil
+		}
+
+		return nil, fmt.Errorf("cannot find network %s of tenant %s: %v", netName, service.Tenant, err)
+	}
+
+	networkUnstructured := network.(*unstructured.Unstructured)
+	var cidrs []string
+	cidrs, ok, err = unstructured.NestedStringSlice(networkUnstructured.Object, "spec", "service", "cidrs")
+	if err != nil {
+		return nil, fmt.Errorf("failed to identify spec.service.cidrs of network %s of tenant %s: %v", netName, service.Tenant, err)
+	}
+	if !ok || len(cidrs) == 0 {
+		return nil, fmt.Errorf("no service cidr found for network network %s of tenant %s", netName, service.Tenant)
+	}
+
+	// todo: to use multiple cidr ranges in the future
+	// for now only first cidr take effect
+	cidr := cidrs[0]
+	var ipNet *net.IPNet
+	if _, ipNet, err = net.ParseCIDR(cidr); err != nil {
+		return nil, fmt.Errorf("%q is not a valid range of network %s of tenant %s: %v", cidr, netName, service.Tenant, err)
```

Comment:
"not a valid range of network" sounds strange.  You mean it's not a valid range of network IPs?

Reply:
I meant for "not a valid service ip range for network object". will reword to make it more clear.